h2/src/main/org/h2/command/dml/Select.java

```diff
@@ -569,7 +569,8 @@ protected ResultInterface queryWithoutCache(int maxRows, ResultTarget target) {
         }
         boolean lazy = session.isLazyQueryExecution() &&
```

Comment:
This is the place where activation/deactivation of lazy evaluation is performed - and as you can see there are many conditions which have to pass before it is activated. I resolved to keep my additional condition here too.